spec/acceptance/class_spec.rb

```diff
@@ -3,22 +3,97 @@
 describe 'agent_upgrade class' do
 
   context 'default parameters' do
-    # Using puppet_apply as a helper
+    before(:all) { setup_puppet }
+    after (:all) { teardown_puppet }
+
     it 'should work idempotently with no errors' do
```

Comment:
I'd have to add an extra test that would be removed when it was no longer pending then. I think updating it is covered sufficiently by [PUP-4606](https://tickets.puppetlabs.com/browse/PUP-4606).